src/source-af-packet.c

```diff
@@ -1877,6 +1909,10 @@ static int AFPCreateSocket(AFPThreadVars *ptv, char *devname, int verbose)
         goto socket_err;
     }
 
+    /* ignore outgoing packets on loopback interfaces */
+    if (if_flags & IFF_LOOPBACK)
+        ptv->pkttype_filter_mask |= BIT_U32(PACKET_OUTGOING);
+
```

Comment:
shouldn't we init to `BIT_U32(PACKET_OUTGOING)|BIT_U32(PACKET_INCOMING)` for the `} else {` case here?

Reply:
pkttype_filter_mask will have a bit set for when the given packet type should be ignored: If loopback, ignore OUTGOING packets, otherwise if no bits are set, all packets are allowed through.

I'm going to reopen v3.